utils/events.ts

```diff
@@ -0,0 +1,103 @@
+import { isBefore, isAfter } from 'date-fns'
+import { toDate } from 'date-fns-tz'
+import { formatUTC, weeksBetween } from './date'
+import { ApiEvent } from 'apiClient'
+
+export type WeeklyData = {
+  week: number
+  date: Date
+  prior: Date
+  events: ApiEvent[]
+}
+
+export const withinBounds = (start: Date, end: Date) => (e: ApiEvent) => {
+  const time = toDate(e.occurred_at)
+  return isAfter(time, start) && isBefore(time, end)
+}
+export const eventsBetween = (
+  start: Date,
+  end: Date,
+  events: ApiEvent[]
+): ApiEvent[] => events.filter(withinBounds(start, end))
+
+export function uniqueById(x: readonly ApiEvent[]) {
+  return x.reduce((agg: ApiEvent[], i: ApiEvent) => {
```

Comment:
Do you want to just build a map with `id` as the key here, then return `.values()`? That seems like it would avoid the need to build an intermediate array and loop over it to check uniqueness.

Reply:
Realized this function isn't actually in use. So I removed it.